### PR TITLE
Add test to create artifact using chunk upload

### PR DESCRIPTION
**Pull request by kersommoura**
Add test to create artifact using chunk upload.

https://pulp.plan.io/issues/4197
ref: #4197

### CODE REVIEW

File: pulpcore/tests/functional/api/using_plugin/test_upload.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables."""
+        cls.cfg = config.get_config()
+        cls.client = api.Client(cls.cfg)
+
+    def test_create_artifact(self):
+        """Test creation of artifact using upload of files in chunks."""
+        file = http_get(FILE_TO_BE_CHUNKED_URL)
```
Comment (daviddavis) [inline pulpcore/tests/functional/api/using_plugin/test_upload.py]: I would move this to `setUpClass` so if a second test gets created it doesn't also try to download the file again.

File: pulpcore/tests/functional/api/using_plugin/test_upload.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        size_file = len(file)
+
+        first_chunk = http_get(FILE_CHUNKED_PART_1_URL)
```
Comment (daviddavis) [inline pulpcore/tests/functional/api/using_plugin/test_upload.py]: Splitting up file in the code would be more efficient. Downloading the file and its chunks downloads twice as much data 20MB vs 10MB.

Comment (kersommoura) [reply pulpcore/tests/functional/api/using_plugin/test_upload.py]: That was my first approach, but the machinery to do this was getting a bit complicated. What about we move forward with this approach, and if it is indeed slow we update?

Comment (daviddavis) [reply pulpcore/tests/functional/api/using_plugin/test_upload.py]: That's fine. At least move this to setUpClass though so we're not downloading the chunks for each upload test

Comment (kersommoura) [reply pulpcore/tests/functional/api/using_plugin/test_upload.py]: That one is already done.

**Comment by kersommoura** [comment]
@bherrin3, I will merge this one. If you have any comment, I will send a follow up PR.

**Pull request merged**